src/test/java/org/prebid/server/auction/PrivacyEnforcementServiceTest.java

```diff
@@ -0,0 +1,615 @@
+        given(gdprService.isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(true));
+        given(gdprService.resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout)))
+                .willReturn(Future.succeededFuture(GdprResponse.of(true, singletonMap(15, true), null)));
+
+        final ExtUser extUser = ExtUser.builder().build();
+        final User user = notMaskedUser();
+        final Device device = notMaskedDevice();
+        final ExtRegs extRegs = ExtRegs.of(1);
+        final Regs regs = Regs.of(null, mapper.valueToTree(extRegs));
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device)
+                        .regs(regs));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, extUser, extRegs, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, regs);
+
+        verify(gdprService).isGdprEnforced(eq("1"), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
+        verify(gdprService).resultByVendor(eq(singleton(15)), eq("1"), any(), any(), eq(timeout));
+        verifyNoMoreInteractions(gdprService);
+
+        assertThat(masked.succeeded()).isTrue();
+        assertThat(masked.result()).hasSize(1)
+                .containsOnly(entry(BIDDER_NAME, expected));
+    }
+
+    @Test
+    public void shouldNotMaskWhenExtDeviceLmtIsNullAndGdprServiceRespondIsGdprEnforcedWithFalse() {
+        // given
+        given(gdprService.isGdprEnforced(isNull(), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout)))
+                .willReturn(Future.succeededFuture(false));
+
+        final User user = notMaskedUser();
+        final Device device = notMaskedDevice();
+        final Map<String, User> bidderToUser = singletonMap(BIDDER_NAME, user);
+
+        final BidRequest bidRequest = givenBidRequest(givenSingleImp(
+                singletonMap(BIDDER_NAME, 1)),
+                bidRequestBuilder -> bidRequestBuilder
+                        .user(user)
+                        .device(device));
+
+        // when
+        final Future<Map<String, PrivacyEnforcementResult>> masked = privacyEnforcementService
+                .mask(bidderToUser, null, null, singletonList(BIDDER_NAME), emptyMap(), bidRequest, PUBLISHER_ID, timeout);
+
+        // then
+        final PrivacyEnforcementResult expected = PrivacyEnforcementResult.of(user, device, null);
+
+        verify(gdprService).isGdprEnforced(isNull(), eq(PUBLISHER_ID), eq(singleton(15)), eq(timeout));
```

Comment:
As I've noted above, if you want to verify what parameters was method called with - use "global" stub with "any()", "anySet()", etc. Otherwise this verification is redundant - if params were different - stub wouldn't be matched and test would've failed with NPE.